gcore/src/errors.rs

```diff
@@ -34,8 +34,9 @@ pub struct SyscallError(pub(crate) u32);
 impl From<SyscallError> for Result<()> {
     fn from(value: SyscallError) -> Self {
         match value.0 {
-            0 => Ok(()),
-            _ => Err(value.get_err()?),
+            gsys::SUCCESS_ERROR_CODE => Ok(()),
+            code => Err(ExtError::decode(code)
```

Comment:
same avoid "decode" here 
```suggestion
            code => Err(ExtError::decode(code)
```